src/ucp/wireup/wireup.c

```diff
@@ -742,6 +742,118 @@ static void ucp_wireup_print_config(ucp_context_h context,
     }
 }
 
+static char* ucp_wireup_add_feature_rsc(ucp_context_h context,
+                                        const ucp_ep_config_key_t *key,
+                                        ucp_lane_map_t lanes_bitmap,
+                                        const char *feature_str,
+                                        char *buf, size_t max)
+{
+    char *p    = buf;
+    char *endp = buf + max;
+    int   sep  = 0;
+    ucp_rsc_index_t rsc_idx;
+    ucp_lane_index_t lane;
+
+    if (!lanes_bitmap) {
+        return p;
+    }
+
+    snprintf(p, endp - p, "%s(", feature_str);
+    p += strlen(p);
+
+    ucs_for_each_bit(lane, lanes_bitmap) {
+       ucs_for_each_bit(rsc_idx, context->tl_bitmap) {
+           if (key->lanes[lane].rsc_index == rsc_idx) {
+               snprintf(p, endp - p, "%*s"UCT_TL_RESOURCE_DESC_FMT, sep, "",
+                        UCT_TL_RESOURCE_DESC_ARG(&context->tl_rscs[rsc_idx].tl_rsc));
+               p += strlen(p);
```

Comment:
q: should it be
```
ret = snprintf(p, end - p, ...);
p += ret;
```
instead of:
```
snprintf(p, endp - p, ...);
p += strlen(p);
```

applicable for all places.

Reply:
discussed offline - nevermind this comment